### PR TITLE
fix(integrations): Upgrade vs Add Another

### PR DESCRIPTION
**Pull request by mnoble**
When both Plugin AND new Integration are installed, we should display "Add Another".

Previously, this wasn't handled and it showed "Upgrade" because the Plugin was installed. This changes that.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx [hunk 1]
```diff
@@ -56,7 +56,10 @@ export default class OrganizationIntegrations extends AsyncComponent {
         i => i.provider.key == provider.key
       );
       const isInstalled = integrations.length > 0;
-      return Object.assign({integrations, isInstalled}, provider);
+      let newProvider = Object.assign({}, provider);
+      newProvider.integrations = integrations;
+      newProvider.isInstalled = isInstalled;
+      return newProvider;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx]: Wait why?

Comment (mnoble) [reply src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx]: :man_shrugging: to be honest. Tests were failing the other way – the `integrations` key wasn't being set.

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx]: was the `isInstalled` key being set? Was `integrations` undefined?

Comment (mnoble) [reply src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx]: `integrations` was an array and definitely had items in it – which is why changing to this dumb syntax fixed things.

Seems like the `Object.assign` + object-shorthand syntax wasn't doing what you'd expect during tests, for whatever reason.

Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx]: the other way means that `provider.integrations` will overwrite the local `integrations`?

I don't know which way is correct, but object spread might be more clear and concise:

```
return {
...provider,
integrations,
isInstalled,
};
```

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx]: Good point, lets just use object spread.

Comment (mnoble) [reply src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx]: Ugh, yea, I had the stupid arguments backwards. I feel it should have still worked. Whatever. Thanks for the JS-fu.

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**